test/base/test_history.py

```diff
@@ -609,27 +609,27 @@ def test_trace_subset(history: pypesto.History):
             partial_trace = getter(arr)
 
             # check partial traces coincide
-            assert len(partial_trace) == len(arr)
+            assert len(partial_trace) == len(arr), var
             for a, b in zip(partial_trace, [full_trace[i] for i in arr]):
                 if var != 'schi2':
-                    assert np.all(a == b) or np.isnan(a) and np.isnan(b)
+                    assert np.all(a == b) or np.isnan(a) and np.isnan(b), var
```

Comment:
[assert_array_equal](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html) provides more helpful output than `==`, but fine to leave, as those were there before.